src/transformStateWithClones.js

```diff
@@ -6,8 +6,45 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArr = [];
+  const createNewObj = (obj) => {
+    const newObj = resultArr.length === 0
+      ? Object.assign({ ...state })
+      : Object.assign({ ...resultArr[resultArr.length - 1] });
+
+    return newObj;
+  };
```

Comment:
It's not a good idea to create functions inside functions. And that depends on something outside the function. The function that creates new objects shouldn't know about "resultArr", it should just create a new object.

Reply:
it's only the way don't repeat this piece of code.  I use it twice